src/data/markdown/translated-guides/en/03 Results output/200 Real-time/00 Prometheus remote write.md

```diff
@@ -31,60 +31,57 @@ All k6 metric types are converted into an equivalent Prometheus remote write typ
 | Counter | Counter                   | `k6_*_total`         |
 | Gauge   | Gauge                     | `k6_*_<unit-suffix>` |
 | Rate    | Gauge                     | `k6_*_rate`          |
-| Trend   | Gauges / Native Histogram | `k6_*_<unit-suffix>` |
+| Trend   | Gauges (default) / [Native Histogram](#prometheus-native-histogram) | `k6_*_<unit-suffix>` |
 
-### Trend metric conversion
+The output maps the metrics into time series with Name labels.
+As much as possible, k6 respects the [naming best practices](https://prometheus.io/docs/practices/naming) that the Prometheus project defines:
+
+* All time series are prefixed with the `k6_` namespace.
+* All time series are suffixed with the base unit of the sample value (if k6 knows what the base unit is).
+* Trends and rates have the relative suffixes, to make them more discoverable.
+
+## Trend metrics and Histograms
+
+By default, Prometheus supports [Counter and Metric types](https://prometheus.io/docs/concepts/metric_types/).  Therefore, this output converts [k6 Trend metrics](https://k6.io/docs/javascript-api/k6-metrics/trend/) to a counter and gauges in Prometheus by default.
 
-By default, k6 trend metrics convert to primitive counters and gauges.
-To convert trend metrics to high-fidelity histograms, you can use Prometheus Native histogram instead.
+Each k6 Trend metric is converted to several Prometheus metrics representing various math functions: count, sum, min, max, avg, med, p(x).
 
-Because k6 can't easily determine fixed buckets in advance, k6 metrics can't export to the original Prometheus histograms.
-So, the output maps a trend metric into primitive counter and gauges.
-Each value represents a math function (count, sum, min, max, avg, med, p(x)).
 This mapping has following drawbacks:
+- Convert a k6 metric to several Prometheus metrics.
 - It is impossible to aggregate some gauge values (especially percentiles).
 - It uses a memory-expensive k6 data structure.
 
-To resolve these limitations, you can map a trend as a [Prometheus Native Histogram](https://prometheus.io/docs/concepts/metric_types/#histogram).
-To do this,
-you can use the `K6_PROMETHEUS_RW_TREND_AS_NATIVE_HISTOGRAM=true` environment variable
-(or one of the other ways).
-The output then converts all the trend types into a dedicated Native Histogram.
+To resolve these limitations, you can convert k6 trend metrics to high-fidelity histograms using Prometheus Native histogram instead.
+
+### Prometheus Native histogram
 
-Mapping trends as native histograms gives you more efficient storage and more precise queries.
-The drawback is that **the feature is experimental, released in Prometheus v2.40.0**.
+A load test could generate vast amounts of data points. Storing raw data is almost always unnecessary and could quickly become expensive and complex to scale.
```

Comment:
Yes, k6 does also other type of testing: browser, functional testing, etc.  But a functional test or browser test are not the ones that generate huge amount of data points. So, storing the raw data of these tests would be ok.

Also, the debate between what is a load test vs a performance test is always there.  But I didn't want to bring this up.

In this context, imo, it is more specific to use the load test term.  Tests with high load are the ones that generates large amount of data and could be problematic.  Initially, I considered using  `A large load test`.

Reply:
>So, storing the raw data of these tests would be ok.

This is the problem to me, you are communicating that Native Histogram is the way to aggregate data and if it is not enabled then you would get raw data. This is not true.
It is crucial to make it clear because there isn't any way for the Prometheus rw output to get raw data. If we don't communicate it clearly then we can get the opposite problem where users start to ask why they are not able to get raw data.

Also, comparing the two methods in terms of data storage will not make the Native Histogram the winner. A Trend converted into a single Gauge (e.g. p99) is probably lighter than the equivalent Native Histogram.

The memory advantage mentioned is on k6 side. But this is a problem with our internal data structure and not related to Prometheus storage, and it will change in the next releases when we have a more efficient data structure.

The Native Histogram's advantages are precision, the ability to decide the math function at query time and not in advance, and the ability to aggregate across multiple time series. None of them is data storage related.